Fix schema errors and some other issues

sean-fitzpatrick opened the pull request:
I pulled all your updates so I could rebuild your book for the upcoming semester.

Running validation detected quite a few new schema violations.

The most common ones:
- You have `<image>` in a `<figure>` because you want to refer to it in an appendix. But a figure **must** have a caption. I added empty captions. This satisfies the schema, but is an indication that perhaps there should be a caption.
- In many sections you added an `<exercise>` *after* the last subsection, but *before* the exercises division. It needs to either be at the end of the subsection, or at the beginning of `<exercises>`. I chose to put it at the beginning of exercises.
- There were still quite a few `<ul>` that were not inside a `<p>` (that was a common issue last time)
- There were several places where there was an introduction, but no subsections that followed. I removed the introduction tags.

If you meant for those exercises to be part of the last subsection, I can move them there. There is also a `<reading-questions>` division you could emply, if that's more what you're thinking of for these.

sean-fitzpatrick committed:
rename to match what is in project.ptx

sean-fitzpatrick committed:
fix schema violations

sean-fitzpatrick committed:
fix ID that I duplicated by mistake

sean-fitzpatrick committed:
add figure to contain xml id, and empty caption

sean-fitzpatrick committed:
can't have two targets named pdf

sean-fitzpatrick committed:
migrate print options from xsl to publication file

sean-fitzpatrick commented:
Also, a note about runestone: I see that you've set up the book for Runestone, but you will not get very good results at the moment. If you want the book to work on Runestone, every video and exercise needs to have a `@label` attribute. This works similar to `xml:id`, but it's used by Runestone to keep track of what activities can be done by a student.

Without labels, Runestone won't add much for your book.

Unfortunately I don't have time to add labels, but you might consider doing so. @bnmnetp might be able to help with this.

sean-fitzpatrick committed:
forgot to remove obsolete lines from color style

sean-fitzpatrick committed:
update import href for print stylesheet

sean-fitzpatrick committed:
and add missed slash. time to quit!

maymk commented:
Sean,

Thanks fro the help.

Things fit in a couple of categories:

1) I decided that it would be helpful to have appendices that summarize the Excel skills.  Honestly, this is mainly for tutors that are helping with a good background in math and great confusion about Excel.  It also makes the first big thing where cross references are used in a significant way.
You mentioned several things that I need to fix in a systematic manner.

2) I am dipping my toes in on including webwork problems in the text.  I figured out how to do this on reading quiz questions which all have the same format and are one question per section.  I would like to add the rest of the webwork questions.  I submitted them to the contrib section of the OPL when we were in Mobile, but the pull request has not been acted on.  Moving them over should be straightforward if it can be done with references and much harder if it needs to be done by adding the webwork code to the pretext code.  For the time being I wanted reading quiz questions kept separate from end of section questions because I need to think about when to allow solutions.  Reading quiz questions are rendered moot if the solution is shown.  Other exercises are more randomized so there is some value to the question even if the student sees a solution with some random key.

3) Porting to Runestone is more complicated than I understood.  I still have work to do.

From your perspective there are several levels in fixing priorities.
A) You are supporting someone at your university using the book.  (Your wife, if I remember correctly.). Do whatever you want to support the instruction at your school.  Part of making something open source, or Creative Commons is that others have a right to modify it to their own needs,

B) You are very generously helping me make improvements on a project for my students.  Your help is greatly appreciated.  However I understand it is volunteer world outside your job description.  Give me a week or so to look at your comments and get myself to the point where I can ask intelligent follow up questions.

C) You have generously helped me with some of the RUNESTONE issues and are providing more suggestions.  Once again, I need to think about what I have been told enough to come up with good followup questions.

There is one more favor I would like to ask.
If I am remembering correctly and your wife will be teaching out of the book, ask her for her preferences on exercises.
Should they stay separate in a webwork setting or go into the text?
Should they have solutions and hints available?
Any other feedback she wants to give.

Thanks again.

Mike

sean-fitzpatrick commented:
I'll certainly ask her for feedback!

I've fixed all issues that I found that could have resulted in glitches for HTML or PDF versions of the book.

The main judgment call I had to make was with the lone exercises: you often had a single lonely exercise that came after the last subsection, but was not part of the exercises division.

I moved it into the exercises. But that may not have been where you want it.

If you review the changes in this pull request, you should be able to get a pretty good idea of what I changed.

The main work that remains for you (or maybe a student, if you can afford to hire one) is to add a label to every exercise and video. The WeBWorK exercises won't work on Runestone until you add labels.

I discovered (with APEX) that if there is more than one video in a section, only one of them will work until you add labels. I don't recall if you ever have two videos in a section.
But even with just one, a label is useful because then the video becomes something you can assign, and Runestone will keep track of which students have actually watched them.
